File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.java
class Solution {
    public List<String> findRepeatedDnaSequences(String s) {
         List<String> res = new ArrayList<>();
         Map<String,Integer> count = new HashMap<>();
         for(int i = 0; i < s.length() ; i++){
            if(!(i+10 <= s.length())) break;
            String seq = s.substring(i,i+10);
            if(count.containsKey(seq)){
                count.put(seq,count.get(seq)+1);
            }
            else{
                count.put(seq,1);
            }
         }
         for(String i : count.keySet()){
            int c=count.get(i);
            if(c > 1) res.add(i);
         } 
         return res ;


            
         


    }
}

File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.py
class Solution(object):
    def findRepeatedDnaSequences(self, s):
        """
        :type s: str
        :rtype: List[str]
        """
        n=len(s)
        result = []
        l=0
        sequence = s[:10]

        d={}
        d[sequence]=1
        
        for r in range(10,n):
            sequence=sequence[1:]+s[r]
            if sequence in d and d[sequence] != -1:
                
                d[sequence]+=1
                if d[sequence] > 1:
                    result.append(sequence)
                    d[sequence] = -1
            elif sequence not in d:
                d[sequence]=1
        return result
            


        














        